test/integration/apiserver/anonymous/anonymous_test.go

```diff
@@ -0,0 +1,229 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package anonymous
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	kubeapiserverapptesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+const (
+	defaultNamespace           = "default"
+	defaultRBACRoleName        = "developer-role"
+	defaultRBACRoleBindingName = "developer-role-binding"
+)
+
+var (
+	defaultRole = &rbacv1.Role{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "Role"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleName},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"list"},
+				Resources:     []string{"pods"},
+				APIGroups:     []string{""},
+				ResourceNames: []string{},
+			},
+		},
+	}
+	defaultRoleBinding = &rbacv1.RoleBinding{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "RoleBinding"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleBindingName},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup: rbac.GroupName,
+				Kind:     rbacv1.UserKind,
+				Name:     "system:anonymous",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "Role",
+			Name:     defaultRBACRoleName,
+		},
+	}
+)
+
+func TestStructuredAuthenticationConfig(t *testing.T) {
+	t.Log("Testing anonymous authenticator with authentication config")
+
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StructuredAuthenticationConfiguration, true)
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.AnonymousAuthConfigurableEndpoints, true)
+
+	testCases := []struct {
+		desc            string
+		authConfig      string
+		additionalFlags []string
+		assertErrFn     func(t *testing.T, errorToCheck error)
+		assertFn        func(t *testing.T, server kubeapiserverapptesting.TestServer)
+	}{
+		{
+			desc: "valid config no conditions",
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				require.NoError(t, errorToCheck)
+			},
+			assertFn: func(t *testing.T, server kubeapiserverapptesting.TestServer) {
+				configureRBAC(t, server)
```

Comment:
RBAC creation isn't instant, do we know the role took effect before we proceed to the attempted gets that depend on the RBAC role? (or put the first get in a retry that tolerates 403 until wait.ForeverTestTimeout)

Reply:
Added a wait. Thanks!

Should we also add the same wait to
https://github.com/kubernetes/kubernetes/blob/f36a821de828372a5f99528d21f309f75e17d043/test/integration/apiserver/oidc/oidc_test.go#L1705?